src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,35 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
+    private final Map<String, UserData> map = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        if (map.containsKey(userId)) {
+            return getUser(userId,executor);
+
+        }
+        try {
+            Thread.sleep(1000);
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+        map.put(userId,new UserData(userId,"some details"));
```

Comment:
When inserting a new 'UserData' into the map, it's important to check if the userId does not already exist in the map to prevent overwriting existing data. This is particularly important in a concurrent environment.